system-test/firestore.js

```diff
@@ -255,6 +255,60 @@ describe('DocumentReference class', function() {
         });
   });
 
+  it('supports arrayUnion()', function() {
```

Comment:
For this test and the one below, maybe add the case where the field is not already an array?

Reply:
I am not 100% opposed, but somewhat lazy. The reason I didn't add these tests is that this is purely server behavior, and nothing in this client would ever influence the type conversion. I am going to leave this untested for now and rely on the backend to do the right thing.